### PR TITLE
Frontend Index.js Refactor

### PR DESCRIPTION
**Pull request by ehsonp**
Edward, Waynar, and I worked on refactoring the index.js file to be much simpler. The router now contains the url and pageType variables, along with the router calling the different setup state functions from the state folder instead of importing it from index. 

Additionally I have a very small frame setup for Ethan and I's log component we are building.

### CODE REVIEW

**Comment by ehsonp** [comment]
Do not merge ran into some bugs already with this refactor give me a day and ill fix these bugs

File: src/front-end/components/log.jsx
Comment (forschoolandnothingelse) [inline src/front-end/components/log.jsx]: You should remove this line since the client will waste time making a network request to get this file, which is just empty.
You should also delete `src/front-end/components/log.css` so nobody tries to use it.

File: src/back-end/schema.js
Comment (forschoolandnothingelse) [inline src/back-end/schema.js]: I just want to make a note that this change means we need to drop the backend again.

**Review by forschoolandnothingelse — CHANGES_REQUESTED** [review]
Just make that one change and make sure this passes linting. I'll test the PR and then get back with an approval once I see it works.

File: src/back-end/schema.js
Comment (ehsonp) [reply src/back-end/schema.js]: Let me discuss this with waynar, i might remove it since it makes more sense to implement in your changes rather than mine. But we agreed this would be a better way to store it

Comment (waynarbocangel) [reply src/back-end/schema.js]: Yeah I'm taking charge of this refactor. Delete it from here please Ehson, thanks!

File: src/front-end/components/log.jsx
Comment (waynarbocangel) [reply src/front-end/components/log.jsx]: This is an incomplete file that he is working on for the next commit. I would ignore it for now since he will be putting stuff in them over the next couple of days but yeah thanks for pointing it out!

File: src/back-end/schema.js
Comment (ehsonp) [reply src/back-end/schema.js]: Will do 
and ill fix linting again

File: src/front-end/components/log.jsx
Comment (forschoolandnothingelse) [reply src/front-end/components/log.jsx]: I think it's worth it to do it for now, since it's always possible things will change and this may end up forgotten about.

Comment (ehsonp) [reply src/front-end/components/log.jsx]: honestly i barely wrote code for it so its ok to remove, i agree it professionally shouldnt be there

**Comment by forschoolandnothingelse** [comment]
I'm going to approve this, but for future use `git rebase <branch>` instead of `git merge <branch>` so we don't have to deal with merge commits.

**Review by forschoolandnothingelse — APPROVED** [review]

**Review by waynarbocangel — APPROVED** [review]
Great work Ehson! Looks like everything split up really nicely.

**Comment by ehsonp** [comment]
Refactored a lot of code in frontend, changed how router and page are updated. State updates display in frontend/state folder. plethora of rebasing issues

**Pull request merged**
